### PR TITLE
Add proj_get_units_from_database() (fixes #2004)

### LINKED ISSUE #2004: proj_list_db_units
Currently there is `proj_list_units` for the old PROJ units. However, there is a WKT based set of units listed in: https://github.com/OSGeo/PROJ/blob/master/data/sql/unit_of_measure.sql

Thoughts on adding something like `proj_list_db_units` or `proj_list_unit_of_measure` ?

### PR DESCRIPTION
**Pull request by rouault**

### CODE REVIEW

File: docs/source/development/reference/functions.rst [hunk 1]
```diff
@@ -521,6 +521,9 @@ Lists
     entry of the returned array is a NULL-entry. The array is statically
     allocated and does not need to be freed after use.
 
+    Note: starting with PROJ 7.1, this function is deprecated by
```
Comment (kbevers) [inline docs/source/development/reference/functions.rst:524]: How about also adding `__attribute__ ((deprecated))` to the function definition in `proj.h` ?

**Review by kbevers — COMMENTED** [review]
Looks good to me

File: docs/source/development/reference/functions.rst (hunk 1 above)
Comment (rouault) [reply docs/source/development/reference/functions.rst:524]: Added

**Review by kbevers — APPROVED** [review]

**Comment by snowman2** [comment]
Just had the thought - what about deprecating `proj_list_angular_units` with this change as well?

**Comment by rouault** [comment]
> what about deprecating `proj_list_angular_units` with this change as well?

==> PR #2079

**Pull request merged**